add async option to standalone urls

wantutopia opened the pull request:
Added code so that dynamic url can be delivered asynchronously even when using plugin in standalone form.

If you create a url asynchronously in vue.config.js and assign it to sitemap.urls, the code is as follows.
```
vue.config.js

pluginOptions: {
    sitemap: {
      pretty: true,
      baseURL: 'https://online.gamecoach.pro',
      defaults: {
        lastmod: new Date().toISOString(),
        changefreq: 'weekly',
        priority: 1.0
      },
      urls: (async () => { return await buildSitemapUrls(); })()
    }
  }
```

At this time, the array wrapped in the Promise object is delivered to the plugin, and the Promise is resolved by awaiting the plugin's options.urls value.

wantutopia committed:
add async option to standalone urls

wantutopia committed:
Update README.md

cheap-glitch commented:
What's preventing you from wrapping your config object in an async IIFE?

```js
(async () => ({
	pluginOptions: {
		sitemap: {
			urls: await buildSitemapUrls()
		}
	}
}))();
```

-----

Side note: this

```js
(async () => { return await buildSitemapUrls(); })()
```
is equivalent to this

```js
buildSitemapUrls()
```

No need to wrap Promises in functions to pass them around ;)

wantutopia committed:
Update README.md

wantutopia commented:
Thanks for the good advice.
In my environment, I had to pass an asynchronous function value to the declarative `module.exports` type in `vue.config.js`.
So, I added the await options on the plugin side.
I've updated the readme with more detail following your advice.
If you think it is unnecessary work, you can close it.
Thanks for the quick feedback. :)
